lwotg/lwotg.go

```diff
@@ -16,3 +23,56 @@ type Hint struct {
 	// Value is the value stored for the hint.
 	Value string
 }
+
+// Server implements the OTG ("Openapi") server.
+type Server struct {
+	*otg.UnimplementedOpenapiServer
+
+	// hintCh is the a channel that is used to send hints about the configuration
+	// to other elements of the OTG system. Particularly, it is used to send hints
+	// to the telemetry mapping functions.
+	hintCh chan Hint
+
+	// protocolHandler is a function called when the OTG SetProtocolState RPC
+	// is called. It is used to ensure that anything that needs to be done in
+	// the underlying system is performed (e.g., ensuring ARP is populated, or
+	// other daemons are started).
+	protocolHandler func(*otg.Config, otg.ProtocolState_State_Enum) error
+
+	// TODO(robjs): Add support for:
+	//   - functions that handle interface configuration.
+	//   - functions that handle protocol configuration.
+	//   - functions that handle flow configuration.
+	//   - functions that handle traffic generation.
+
+	// cfg is a cache of the current OTG configuration.
+	cfg *otg.Config
+}
+
+// New returns a new lightweight OTG (LWOTG) server.
```

Comment:
meta golang question - what's the best pattern for where to put "initialization" arguments? For example, why not take the protocolhandler in this New() func, if it's needed for the server to function?

Reply:
Good question. I think this is really just a code design question rather than necessarily one that is golang-specific. In this case, one could have a `nil` protocol handler -- if you were just wanting to have the OTG server come up and do some configuration qualification against it. Given that this is possible (but maybe not likely) then I figured it was better to have this be added at a later point.

There were two other things here that drove this choice:

* How the caller's code reads -- if the caller has to provide all of the arguments to the `New` function, then they end up writing code that declares them all up-front before then creating the server - which seems less readable to me than creating the server and then saying "but add this function".
* Extensibility -- if the `New` is as minimal as possible (but no more minimal :-)) then this means when we decide to add something else, we don't necessarily need to do that in a way that is backwards incompatible.